jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -589,8 +589,8 @@ static List<HttpCookie> getCookies(Request request)
         CookieCache cookieCache = (CookieCache)request.getComponents().getCache().getAttribute(CACHE_ATTRIBUTE);
         if (cookieCache == null)
         {
-            // TODO compliance listeners?
-            cookieCache = new CookieCache(request.getConnectionMetaData().getHttpConfiguration().getRequestCookieCompliance(), null);
+            // TODO: CookieCache is per connection, needs to be per request for ComplianceViolation.Listener
+            cookieCache = new CookieCache(request.getConnectionMetaData().getHttpConfiguration().getRequestCookieCompliance(), HttpChannel.from(request).getComplianceViolationListener());
```

Comment:
This needs to be fixed.  I think we should throw or return a ComplianceViolationException rather than pass in the listeners.

Reply:
done